src/rotation.rs

```diff
@@ -0,0 +1,645 @@
+    let rm90 = Rotation3D::around_z(Radians::new(-FRAC_PI_2));
+    let r180 = Rotation3D::around_z(Radians::new(PI));
+
+    assert!(ri.transform_point2d(&point2(1.0, 2.0)).approx_eq(&point2(1.0, 2.0)));
+    assert!(r90.transform_point2d(&point2(1.0, 2.0)).approx_eq(&point2(-2.0, 1.0)));
+    assert!(rm90.transform_point2d(&point2(1.0, 2.0)).approx_eq(&point2(2.0, -1.0)));
+    assert!(r180.transform_point2d(&point2(1.0, 2.0)).approx_eq(&point2(-1.0, -2.0)));
+
+    assert!(
+        r90.inverse().inverse().transform_point2d(&point2(1.0, 2.0)).approx_eq(
+            &r90.transform_point2d(&point2(1.0, 2.0))
+        )
+    );
+}
+
+#[test]
+fn pre_post() {
+    use std::f32::consts::{FRAC_PI_2};
+    let r1 = Rotation3D::around_x(Radians::new(FRAC_PI_2));
+    let r2 = Rotation3D::around_y(Radians::new(FRAC_PI_2));
+    let r3 = Rotation3D::around_z(Radians::new(FRAC_PI_2));
+
+    let t1 = r1.to_transform();
+    let t2 = r2.to_transform();
+    let t3 = r3.to_transform();
+
+    let p = point3(1.0, 2.0, 3.0);
+
+    // Check that the order of transformations is correct (corresponds to what
+    // we do in Transfor3D).
+    let p1 = r1.post_rotate(&r2).post_rotate(&r3).transform_point3d(&p);
+    let p2 = t1.post_mul(&t2).post_mul(&t3).transform_point3d(&p);
+
+    assert!(p1.approx_eq(&p2));
+
+    // Check that changing the order indeed matters.
+    let p3 = t3.post_mul(&t1).post_mul(&t2).transform_point3d(&p);
+    assert!(!p1.approx_eq(&p3));
+}
+
+#[test]
+fn to_transform3d() {
+    use std::f32::consts::{PI, FRAC_PI_2};
+    let rotations = [
+        Rotation3D::identity(),
+        Rotation3D::around_x(Radians::new(FRAC_PI_2)),
+        Rotation3D::around_x(Radians::new(-FRAC_PI_2)),
+        Rotation3D::around_x(Radians::new(PI)),
+        Rotation3D::around_y(Radians::new(FRAC_PI_2)),
+        Rotation3D::around_y(Radians::new(-FRAC_PI_2)),
+        Rotation3D::around_y(Radians::new(PI)),
+        Rotation3D::around_z(Radians::new(FRAC_PI_2)),
+        Rotation3D::around_z(Radians::new(-FRAC_PI_2)),
+        Rotation3D::around_z(Radians::new(PI)),
+    ];
+
+    let points = [
+        point3(0.0, 0.0, 0.0),
+        point3(1.0, 2.0, 3.0),
+        point3(-5.0, 3.0, -1.0),
+        point3(-0.5, -1.0, 1.5),
+    ];
+
+    for rotation in &rotations {
+        for point in &points {
+            let p1 = rotation.transform_point3d(point);
+            let p2 = rotation.to_transform().transform_point3d(point);
+            assert!(p1.approx_eq(&p2));
+        }
+    }
+}
+
+#[test]
+fn slerp() {
+    let q1 = Rotation3D::quaternion(1.0, 0.0, 0.0, 0.0);
+    let q2 = Rotation3D::quaternion(0.0, 1.0, 0.0, 0.0);
+
+    // The values below can be obtained with a python program:
+    // import numpy
+    // import quaternion
+    // q1 = numpy.quaternion(1, 0, 0, 0)
+    // q2 = numpy.quaternion(0, 1, 0, 0)
+    // quaternion.slerp_evaluate(q1, q2, .0)
+
+    assert!(q1.slerp(&q2, 0.0).approx_eq(&q1));
```

Comment:
would be nice to also have tests for the code paths of A) quaternions being very close to each other, and B) quaternions in different hemispheres